bash_completion

```diff
@@ -363,10 +366,92 @@ _comp_split()
     local _original_opts=$SHELLOPTS
     set -o noglob
 
-    eval "$1$_assign(\$2)"
+    local _old_size _new_size
+    if "$_append"; then
+        eval "$1+=()" # in case $1 is unset
+        eval "_old_size=\${#$1[@]}"
+        eval "$1+=(\$2)"
+    else
+        _old_size=0
+        eval "$1=(\$2)"
+    fi
+    eval "_new_size=\${#$1[@]}"
 
     [[ :$_original_opts: == *:noglob:* ]] || set +o noglob
-    return 0
+    ((_new_size > _old_size))
+}
+
+# Call `compgen` with the specified arguments and store the results in the
+# specified array.
+# Usage: _comp_compgen [-al] [-F sep] ARRAY args...
+# This function essentially performs ARRAY=($(compgen args...)) but properly
+# handles shell options, IFS, etc. using _comp_split.  This function is
+# equivalent to `_comp_split [-a] -F sep ARRAY "$(compgen args...)"`, but this
+# pattern is frequent in the codebase and is good to separate out as a function
+# for the possible future implementation change.
+# OPTIONS
+#   -a      Append to the array
+#   -F sep  Set a set of separator characters (used as IFS in evaluating
+#           `compgen').  The default separator is $' \t\n'.  Note that this is
+#           not the separators to delimit the output of `compgen' but the
+#           separators in evaluating the expansions of `-W '...'`, etc.  The
+#           delimiter of the output of `compgen` is always a newline.
+#   -l      The same as -F $'\n'
+# @param $1 array_name  The array name
+#   The array name should not start with an underscores "_", which is
+#   internally used.  The array name should not be either "IFS" or
+#   "OPT{IND,ARG,ERR}".
+# @param $2... args     The arguments that are passed to compgen
+#   Note: References to positional parameters $1, $2, ... (such as -W '$1')
+#   will not work as expected because these references the arguments of
+#   `_comp_compgen' instead of those of the caller function.  When there are
+#   needs to reference them, save the arguments to an array and reference the
+#   array instead.
+_comp_compgen()
+{
+    local _append=false IFS=$' \t\n'
+    local -a _split_options=(-l)
+
+    local OPTIND=1 OPTARG="" OPTERR=0 _opt
+    while getopts ':alF:' _opt "$@"; do
+        case $_opt in
+            a) _append=true _split_options+=(-a) ;;
+            l) IFS=$'\n' ;;
+            F) IFS=$OPTARG ;;
+            *)
+                echo "bash_completion: $FUNCNAME: usage error" >&2
+                return 2
+                ;;
+        esac
+    done
+    shift "$((OPTIND - 1))"
+    if (($# < 2)); then
```

Comment:
Maybe?
```suggestion
    if (($# != 2)); then
```

Reply:
We expect two or more arguments here. We need an array name plus arguments that will be specified to `compgen`.

Reply:
I realized my mistake in the usage text. I think I copied the related code from `_comp_split` but forgot to update the usage text. I now fixed it in 0a267dc2bf74e9956aacc2b31d032d4d239814d6.